File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/legacy/ml/BarcodeScannerOptions.kt
package dev.hebei.camerax_android.legacy.ml

import dev.hebei.camerax_android.legacy.BarcodeFormat
import dev.hebei.camerax_android.legacy.CameraXRegistrar
import dev.hebei.camerax_android.legacy.PigeonApiBarcodeScannerOptions

class BarcodeScannerOptions(registrar: CameraXRegistrar) : PigeonApiBarcodeScannerOptions(registrar) {
    override fun build(
        enableAllPotentialBarcodes: Boolean?,
        formats: List<BarcodeFormat>?,
        zoomSuggestionOptions: com.google.mlkit.vision.barcode.ZoomSuggestionOptions?
    ): com.google.mlkit.vision.barcode.BarcodeScannerOptions {
        val builder = com.google.mlkit.vision.barcode.BarcodeScannerOptions.Builder()
        if (enableAllPotentialBarcodes == true) {
            builder.enableAllPotentialBarcodes()
        }
        if (!formats.isNullOrEmpty()) {
            val format = formats.first().obj
            val moreFormats = formats.drop(1).map { it.obj }.toIntArray()
            builder.setBarcodeFormats(format, *moreFormats)
        }
        if (zoomSuggestionOptions != null) {
            builder.setZoomSuggestionOptions(zoomSuggestionOptions)
        }
        return builder.build()
    }
}

File: camerax/lib/src/ml/barcode/barcode_scanner_options.dart
import 'package:camerax_platform_interface/camerax_platform_interface.dart'
    as $interface;
import 'package:meta/meta.dart';

import 'zoom_suggestion_options.dart';

final class BarcodeScannerOptions {
  final $interface.BarcodeScannerOptions _obj;

  BarcodeScannerOptions({
    bool? enableAllPotentialBarcodes,
    List<$interface.BarcodeFormat>? formats,
    ZoomSuggestionOptions? zoomSuggestionOptions,
  }) : _obj = $interface.BarcodeScannerOptions(
          enableAllPotentialBarcodes: enableAllPotentialBarcodes,
          formats: formats,
          zoomSuggestionOptions: zoomSuggestionOptions?.obj,
        );

  @internal
  $interface.BarcodeScannerOptions get obj => _obj;
}
